web/satellite/src/components/team/HeaderArea.vue

```diff
@@ -164,6 +175,35 @@ export default class HeaderArea extends Vue {
     public get areSelectedProjectMembersBeingDeleted(): boolean {
         return this.headerState === 1 && this.isDeleteClicked;
     }
+
+    private async deleteWithCurrentUser(): Promise<void> {
+        const selectedProject = this.$store.getters.selectedProject;
+        try {
+            await this.$store.dispatch(PM_ACTIONS.DELETE);
+            await this.$store.dispatch(PROJECTS_ACTIONS.DELETE_STORED_PROJECT, selectedProject.id);
+
+            await this.setProjectState();
+        } catch (error) {
+            await this.$notify.error(error.message);
+            this.isDeleteClicked = false;
+
+            return;
+        }
+    }
+
+    private async setProjectState(): Promise<void> {
+        const projects = await this.$store.dispatch(PROJECTS_ACTIONS.FETCH);
+        if (projects.length) {
```

Comment:
if (!projects.length) {
        await this.$store.dispatch(APP_STATE_ACTIONS.CHANGE_STATE, AppState.LOADED_EMPTY);
        await this.$router.push(RouteConfig.ProjectOverview.with(RouteConfig.ProjectDetails).path);
            
            return;
        }
            await this.$store.dispatch(PROJECTS_ACTIONS.SELECT, projects[0].id);
            await this.$store.dispatch(PM_ACTIONS.FETCH, this.FIRST_PAGE);
            this.$refs.headerComponent.clearSearch();